src/modalities/dataloader/dataloader.py

```diff
@@ -37,7 +37,7 @@ def __init__(
             batch_sampler (ResumableBatchSampler): The batch sampler used for sampling batches.
             dataset (Dataset[T_co]): The dataset to load the data from.
             batch_size (Optional[int], optional): The number of samples per batch. Defaults to 1.
-            sampler (Union[Sampler, Iterable, None], optional): The sampler used for sampling data. Defaults to None.
+            sampler (Sampler | Iterable | None, optional): The sampler used for sampling data. Defaults to None.
```

Comment:
```suggestion
            sampler (Optional[Sampler | Iterable], optional): The sampler used for sampling data. Defaults to None.
```
See lines 18 & 39 (one could also remove the "optional" in lines 39 & 40 as they are / become redundant if "Optional" is used)

Reply:
That means you suggest changing it to:
```python
batch_size (Optional[int]): The number of samples per batch. Defaults to 1.
sampler (Optional[Sampler | Iterable]): The sampler used for sampling data. Defaults to None.
```
right?

Reply:
Yes. But in that case, the "original" should probably also be removed from line 39, for the sake of consistency.

Reply:
I am not quite sure I understand what you mean. If you are talking about removing "optional" from line 39 then that's what I did in the example. Feel free to correct me if I am wrong.

Reply:
Sorry, I simply missed that you changed line 39 as well in your above suggestion. Looks good.